app/src/test/java/org/simple/clinic/patient/filter/SortByTotalSumOfDistancesTest.kt

```diff
@@ -0,0 +1,69 @@
+package org.simple.clinic.patient.filter
+
+import com.google.common.truth.Truth.assertThat
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.patient.PatientSearchResult.PatientNameAndId
+import java.util.UUID
+
+@RunWith(JUnitParamsRunner::class)
+class SortByTotalSumOfDistancesTest {
+
+  @Test
+  @Parameters(method = "params for sorting values")
+  fun `it should sort the values based on the total sum of the distances`(
+      originalList: List<SearchInfo>,
+      expected: List<SearchInfo>
+  ) {
+    val sorted = originalList.sortedWith(SortByTotalSumOfDistances())
+
+    assertThat(sorted).isEqualTo(expected)
+  }
+
+  @Suppress("Unused")
+  private fun `params for sorting values`(): List<List<Any>> {
+    fun generateTestData(distances: List<List<Double>>, results: List<Int>): List<Any> {
+      val originalList = distances.map {
+        SearchInfo(
+            patient = PatientNameAndId(fullName = "Name", uuid = UUID.randomUUID()),
+            nameParts = emptyList(),
+            searchParts = emptyList(),
+            distances = it
+        )
+      }
+      val expected = results.map { originalList[it] }
+
+      return listOf(originalList, expected)
+    }
+
+    return listOf(
+        generateTestData(
+            distances = listOf(
+                listOf(1.0),
```

Comment:
Will using local data classes here instead of using numbers directly make this function more understandable? I see lots of numbers here.

Reply:
It might. I'll take a stab at it.